src/mustache.rs

```diff
@@ -50,6 +50,22 @@ impl Mustache {
         .collect(),
     }
   }
+
+  pub fn all_parts(&self) -> Vec<&Vec<String>> {
```

Comment:
```suggestion
  pub fn segments(&self) -> Option<&Vec<String>> {
```

I think would be a better function signature.

Reply:
Hm, making the return type an `Option` of a `Vec` won't work here since we're collecting all the segments (`Vec`) into another `Vec`, not just returning one of the inner segments

Reply:
How would this work for an expression - `{{a.b}}c{{d.e}}`?

Reply:
yes, that is intentional since i just indented for this to be a function to get the segments we care about for validation, so that i dont have to spam this pattern in multiple places where you `match` `self` with `Mustache` and then get the inner `segments`

Reply:
renamed the function to be more explicit regarding this